riscv/processor.cc

```diff
@@ -886,6 +889,14 @@ void processor_t::take_trap(trap_t& t, reg_t epc)
     vsdeleg = (curr_virt && state.prv <= PRV_S) ? (state.medeleg->read() & state.hedeleg->read()) : 0;
     hsdeleg = (state.prv <= PRV_S) ? state.medeleg->read() : 0;
   }
+  // An unexpected trap - a trap when SDT is 1 - traps to M-mode
+  if ( (state.prv <= PRV_S && bit < max_xlen) &&
+       (((vsdeleg >> bit) & 1)  || ((hsdeleg >> bit) & 1)) ) {
+    reg_t s = curr_virt ? state.nonvirtual_sstatus->read() : state.sstatus->read();
+    supv_double_trap = get_field(s, MSTATUS_SDT);
```

Comment:
If `supv_double_trap` is only used in this scope, delete its declaration above and instead declare it as part of assigning it here.  (Or get rid of the variable altogether and just write `if (get_field(s, MSTATUS_SDT))` below.)

Reply:
Oops, nevermind, I see it is used below.  Ignore.